test/e2e/functional_test.go

```diff
@@ -838,6 +838,37 @@ spec:
 		WaitForWorkflow(fixtures.ToBeSucceeded)
 }
 
+func (s *FunctionalSuite) TestPauseBefore() {
```

Comment:
how long does this test take please?

Reply:
I know we want to avoid using `sleep` but as I understand I have to add some type of sleep in order to make sure that the main container is up and running.  If you still think the sleep is an issue I will take a look at adding a loop to the bash to continue retry the command during 5 seconds instead to cut some time. WDYT @alexec ?

These are the times from local run: 
```
    --- PASS: TestFunctionalSuite/TestPauseAfter (8.03s)
    --- PASS: TestFunctionalSuite/TestPauseAfterAndBefore (8.02s)
    --- PASS: TestFunctionalSuite/TestPauseBefore (8.06s)
```

Reply:
I think 24s is ok